src/WorldIDIdentityManagerImplV3.sol

```diff
@@ -0,0 +1,233 @@
+    ///      replaced by use of setters.
+    /// @dev This function is explicitly not virtual as it does not make sense to override even when
+    ///      upgrading. Create a separate initializer function instead.
+    ///
+    ///
+    /// @custom:reverts InvalidVerifierLUT if `_batchInsertion4844Verifiers` is set to the zero address
+    function initializeV3(VerifierLookupTable _batchInsertion4844Verifiers) public reinitializer(3) {
+        if (address(_batchInsertion4844Verifiers) == address(0)) {
+            revert InvalidVerifierLUT();
+        }
+
+        batchInsertion4844Verifiers = _batchInsertion4844Verifiers;
+
+        emit WorldIDIdentityManagerImplV3Initialized();
+    }
+
+    ///////////////////////////////////////////////////////////////////
+    ///                     IDENTITY MANAGEMENT                     ///
+    ///////////////////////////////////////////////////////////////////
+
+    /// @notice Parameters for registerIdentities function.
+    /// @dev This struct holds the parameters for registering identities and verifying the insertion proof
+    ///      using KZG proofs as described in EIP-4844.
+    struct RegisterIdentities4844Params {
+
+        /// @notice The proof that given the conditions, insertion into the tree results in `postRoot`.
+        ///         Elements 0 and 1 are the `x` and `y` coordinates for `ar` respectively. Elements 2 and 3 are the
+        ///         `x` coordinate for `bs`, and elements 4 and 5 are the `y` coordinate for `bs`. Elements 6 and 7
+        ///         are the `x` and `y` coordinates for `krs`.
+        uint256[8] insertionProof;
+
+        /// @notice The Pedersen commitments from the proof.
+        uint256[2] commitments;
+
+        /// @notice The proof of knowledge for the Pedersen commitments.
+        uint256[2] commitmentPok;
+
+        /// @notice KZG commitment for the polynomial extrapolated from the identities.
+        uint128[3] kzgCommitment;
+
+        /// @notice KZG proof associated with the commitment.
+        uint128[3] kzgProof;
+
+        /// @notice Expected evaluation of the polynomial at a certain point equal to kzgChallenge.
+        uint256 expectedEvaluation;
+
+        /// @notice The value for the root of the tree before the `identityCommitments` have been inserted.
+        ///         Must be an element of the field `Kr`. (already in reduced form)
+        uint256 preRoot;
+
+        /// @notice The root obtained after inserting all of `identityCommitments` into the tree described
+        ///         by `preRoot`. Must be an element of the field `Kr`. (already in reduced form)
+        uint256 postRoot;
+
+        /// @notice Challenge value used in the KZG proof to evaluate the polynomial.
+        uint256 kzgChallenge;
+
+        /// @notice Hash of the input data calculated as described in the comment
+        ///         to `calculateIdentityRegistrationInputHash()`.
```

Comment:
No, this is not how this is calculated. The contract never actually computes this.